runtime/drivers/postgres/sql_store.go

```diff
@@ -0,0 +1,190 @@
+package postgres
+
+import (
+	"context"
+	sqldriver "database/sql/driver"
+	"fmt"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime/drivers"
+	"golang.org/x/exp/slices"
+
+	// load pgx driver
+	_ "github.com/jackc/pgx/v4/stdlib"
+)
+
+// Query implements drivers.SQLStore
+func (c *connection) Query(ctx context.Context, props map[string]any) (drivers.RowIterator, error) {
```

Comment:
The `SourceProperties` make it seem like the `database_url` can be overriden in source props, but it's not consumed here (only in `Open`, which gets `ConfigProperties`)

Reply:
Yeah that is one different implementation from other connectors. Everywhere else we are not opening any connection in `drivers.Open` at least for source connectors. May be I can keep the implementation same in postgres as well and only open connection to underlying infra in `Query` method. This would also make it easy to handle this comment : https://github.com/rilldata/rill/pull/3065#discussion_r1325080034
What do you think ?

Reply:
Seems okay to open it in `Query`. I wonder if we should remove the `database_url` from the source properties though, and require people to configure it as connector properties? That's how we do it for other connectors.